app/controllers/concerns/blacklight/controller.rb

```diff
@@ -133,6 +133,9 @@ def current_or_guest_user
     def discard_flash_if_xhr
       flash.discard if request.xhr?
     end
+    deprecation_deprecate discard_flash_if_xhr: "Discarding flash messages on XHR requests is deprecated.
```

Comment:
What if you don't want to continue with the behavior, but you just want the deprecation to go away? Should we add `skip_after_action :discard_flash_if_xhr`?

Reply:
Yep, you would add a skip. Added these instructions to the warning.